syntax = "proto2";

package configrequest;

message Request {
	enum RequestType {
		IFACE_INFO = 0;
		ROUTER_CONF = 1;
		IP_ROUTES = 2;
		START_XIA = 3;
		GATHER_XIDS = 4;
	}
		
	message InterfaceInfo {
		required string name = 1;
		optional string ipaddr = 2;
		optional string macaddr = 3;
	}

	message GetInterfaceInfo {
		repeated InterfaceInfo interfaces = 1;
	}

	message DeployRouterConf {
		required string configfile = 1;
	}

	message SetIPRoutes {
		repeated string route_cmds = 1;
	}

	message StartXIA {
		required string command = 1;
		optional string resolvconf = 2;
	}

	message GatherXIDs {
		optional string ad = 1;
		optional string hid = 2;
	}

	required RequestType type = 1;
	optional GetInterfaceInfo ifrequest = 2;
	optional DeployRouterConf routerconf = 3;
	optional SetIPRoutes routes = 4;
	optional StartXIA startxia = 5;
	optional GatherXIDs gatherxids = 6;
}
